plugin-server/src/kafka/error-handling.ts

```diff
@@ -0,0 +1,54 @@
+import { DependencyUnavailableError } from '../utils/db/error'
+import { status } from '../utils/status'
+
+export const retryOnDependencyUnavailableError = async <T>(
+    fn: () => Promise<T>,
+    {
+        retryCount,
+        initialRetryDelayMs,
+    }: {
+        retryCount: number
+        initialRetryDelayMs: number
+    } = { retryCount: 5, initialRetryDelayMs: 1000 }
+): Promise<T> => {
+    // Try processing the message. If we get a
+    // DependencyUnavailableError retry up to 5 times starting
+    // with a delay of 1 second, then 2 seconds, 4 seconds, 8
+    // seconds, and finally 16 seconds. If we still get an error
+    // after that, we will throw it and stop processing.
+    // If we get any other error, we will throw it and stop
```

Comment:
question: this means that the pod will sit idle, but will eventually fail its liveness and get restarted?
We need to be able to weather a broker upgrade without paging on-call.

Reply:
There are no Kafka errors thrown from this function I don't _think_ although that could be made more clear.

It's only the await on the produce promises that will raise. Let me check what the default retries are for the producer when we have retriable errors e.g. broker upgrades.